lib/improver/cli/nowcast_accumulate.py

```diff
@@ -0,0 +1,117 @@
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to accumulate input data given advection velocity fields."""
+
+from improver import cli
+
+inputadvection = cli.create_constrained_inputcube_converter(
+    'precipitation_advection_x_velocity', 'precipitation_advection_y_velocity')
+
+
+@cli.clizefy
+@cli.with_output
+def process(input_cube: cli.inputcube, advection_cubes: inputadvection,
+            *oe_cube: cli.inputcube, attributes_dict: cli.inputjson = None,
+            max_lead_time=360, lead_time_interval=15, accumulation_period=15,
+            accumulation_units='m'):
+    """Module to extrapolate and accumulate the weather with 1 min fidelity.
+
+    Args:
+        input_cube (iris.cube.Cube):
+            The input Cube to be processed.
+        advection_cubes (iris.cube.CubeList):
+            Advection cubes of U and V.
+        oe_cube (iris.cube.Cube):
+            Cube containing the orographic enhancement fields. May have data
+            for multiple times in the cube.
+            Default is None.
+        attributes_dict (dict):
+            Dictionary containing the required changes to the attributes.
+            Default is None.
+        max_lead_time (int):
+            Maximum lead time required (mins).
+            Default is 360.
+        lead_time_interval (int):
+            Interval between required lead times (mins).
+            Default is 15.
+        accumulation_period (int):
+            The period over which the accumulation is calculated (mins).
+            Only full accumulation periods will be computed. At lead times
+            that are shorter than the accumulation period, no accumulation
+            output will be produced.
+        accumulation_units (str):
+            Desired units in which the accumulations should be expressed.
+            e.g. 'mm'
+            Default is 'm'.
+
+    Returns:
+        iris.cube.CubeList:
+            New cubes with accumulated data.
+
+    Raises:
+        ValueError:
+            can either use speed_cube and direction_cube or u_cube and v_cube.
```

Comment:
Mistake here? Seems to raise if either u or v are missing.

Reply:
`If advection_cubes doesn't contain an x velocity and a y velocity.`